### ISSUE TITLE
riak_repl2_rtq (possibly) consumes large memory

### ISSUE THREAD

**Issue by shino** [basho/riak_repl#258]
When site is down or not configured, riak_repl2_rtq (possibly) consumes large memory.

There is setting item for max bytes [1], but only ETS table size is checked [2].
Because Riak CS uses 1MB chunk size, riak_repl2_rtq may hold more than 100 GB memory.

[1] https://github.com/basho/riak_repl/blob/1.3/src/riak_repl2_rtq.erl#L114
[2] https://github.com/basho/riak_repl/blob/1.3/src/riak_repl2_rtq.erl#L386

Environment:
- riak_ee.git branch 1.3 (ad47b04d14cb30adcf221d285b32b0346cbc67b9)
- Ubuntu 12.10, amd64
- Erlang/OTP R15B03-1

steps to reporoduce:

```
$ make stagedev1
$ dev/dev1/bin/riak start
$ dev/dev1/bin/riak-repl add-listener dev1@127.0.0.1 127.0.0.1 11016
$ dd if=/dev/zero of=zero.1MB bs=1M count=1

$ for i in {1..300}; do
    curl -X PUT --data-binary @zero.1MB http://127.0.0.1:10018/buckets/test/keys/${i};
  done

$ dev/dev1/bin/riak attach
> [erlang:garbage_collect(P)|| P <- erlang:processes()].
> erlang:memory(binary).
314835224  %% ~ 300MB
> [erlang:external_size(M) || M <- riak_repl2_rtq:dumpq()].
[1049120,1049122,1049122,1049122,1049122,1049122,1049120,
 1049122,1049122,1049123,1049123,1049123,1049123,1049123,
 1049123,1049123,1049123,1049123,1049123,1049123,1049123,
 1049123,1049123,1049121,1049123,1049121,1049123,1049123,
 1049121|...]

> lists:sum([erlang:external_size(M) || M <- riak_repl2_rtq:dumpq()]).
314737151  %% ~ 300 MB

> ets:i().
 id              name              type  size   mem      owner
 ----------------------------------------------------------------------------
 [snip]
 983210          riak_repl2_rtq    ordered_set 300    4590     riak_repl2_rtq
 [snip]

> 4590*erlang:system_info(wordsize)/300.
122.4   %% bytes per riak_repl2_rtq ETS record
```

**Comment by jonmeredith** [comment]
Thank you for the excellent demonstration Shino.  We need to address this.

**Comment by bookshelfdave** [comment]
Fixed in #261. 

RTQ EQC left as-is until the cascading writes changes are merged, as that PR put a lot of work into the EQC test.
